docs/src/components/ReactExample/transform.ts

```diff
@@ -1,95 +1,140 @@
 import { transform as babelTransform } from "@babel/standalone";
-import _ from "lodash";
-import type { BabelFileResult } from "@babel/core";
+import type { BabelFileResult, NodePath } from "@babel/core";
+import type {
+  ObjectProperty,
+  JSXAttribute,
+  JSXIdentifier,
+  JSXOpeningElement,
+  JSXSpreadAttribute,
+} from "@babel/types";
+import fp from "lodash/fp";
 
-/* eslint-disable no-param-reassign */
+const parseArrayOfObjectProps = (props: ObjectProperty[]) => {
+  return props.reduce((acc, prop) => {
+    if (prop.type === "ObjectProperty") {
+      const { key, value } = prop;
+      if (key.type === "Identifier") {
+        if (
+          value.type === "NumericLiteral" ||
+          value.type === "StringLiteral" ||
+          value.type === "BooleanLiteral"
+        ) {
+          acc[key.name] = value.value;
+        }
+      }
+    }
+
+    return acc;
+  }, {});
+};
 
 export const transform = (
   exampleName: string,
-  example: string,
+  code: string,
   knobs: Record<string, Record<string, string | number | boolean>>,
 ) => {
-  const result: BabelFileResult = babelTransform(example, {
+  const result: BabelFileResult = babelTransform(code, {
     filename: exampleName,
     sourceType: "module",
-    presets: ["typescript", "react"],
     plugins: [
+      "syntax-jsx",
       ({ types: t }) => {
-        const getAttribute = (value: string | number | boolean, name: string) => {
-          const stringified = value.toString();
-
-          if (stringified === "true") return t.jsxAttribute(t.jsxIdentifier(name), null);
-          if (stringified === "false")
-            return t.jsxAttribute(
-              t.jsxIdentifier(name),
-              t.jsxExpressionContainer(t.booleanLiteral(false)),
-            );
-
-          if (typeof value === "number") {
-            return t.jsxAttribute(
-              t.jsxIdentifier(name),
-              t.jsxExpressionContainer(t.numericLiteral(value)),
+        const parsePropValue = (value: string | number | boolean) => {
+          if (typeof value === "string" && value.includes("-icon")) {
+            const iconName = `${value.split("-icon")[0]}`;
+            return t.jsxExpressionContainer(
+              t.jsxElement(
+                t.jsxOpeningElement(
+                  t.jsxMemberExpression(t.jsxIdentifier("Icons"), t.jsxIdentifier(iconName)),
+                  [],
+                  true,
+                ),
+                null,
+                [],
+                true,
+              ),
             );
           }
 
-          return t.jsxAttribute(t.jsxIdentifier(name), t.stringLiteral(stringified));
+          if (typeof value === "boolean") return t.jsxExpressionContainer(t.booleanLiteral(value));
+          if (typeof value === "number") return t.jsxExpressionContainer(t.numericLiteral(value));
+          if (typeof value === "string") return t.stringLiteral(value);
+
```

Comment:
I remember we hardcoded 🔨  this. Is there no turning back?

Reply:
We will solve this in another PR, that's outside of the scope of this, currently, controlled/uncontrolled components also do not work correctly :)